File: 1/1-4.ts
function p(): number {
    return p();  // Infinite recursion
}

function test(x: number, y: number): number {
    if (x === 0) {
        return 0;
    } else {
        return y;
    }
}

// Uncommenting this will cause an infinite loop or stack overflow
// console.log(test(0, p()));

File: 1/1-4.rs
fn p() -> i32 {
    p()  // Infinite recursion
}

fn test(x: i32, y: i32) -> i32 {
    if x == 0 {
        0
    } else {
        y
    }
}

fn main() {
    // Uncommenting this will cause an infinite loop or stack overflow
    // println!("{}", test(0, p()));
}
